test.rb

```diff
@@ -0,0 +1,406 @@
+             chemistry_average=(chemistry_total)/(@hash_array.length-count)
+             htemp_year={ :year => @hash_array[@hash_array.length-1][:year], :maths => math_average, :physics => physics_average , :chemistry => chemistry_average }
+            @hash_array_groupby.push(htemp_year)
+
+        else
+          puts "Enter Some valid input"
+       end
+
+    end
+     # Sort_by method
+       def sort_hashes(sort_by)
+          if(sort_by.downcase=="maths")
+             @hash_array_groupby= @hash_array_groupby.sort_by { |hsh| hsh[:maths] }.reverse
+
+          elsif (sort_by.downcase=="physics")
+            @hash_array_groupby= @hash_array_groupby.sort_by { |hsh| hsh[:physics] }.reverse
+
+          elsif(sort_by.downcase=="chemistry")
+            @hash_array_groupby= @hash_array_groupby.sort_by { |hsh| hsh[:chemistry] }.reverse
+          else
+             puts "Enter Some valid input"
+
+        end
+       end
+
+    def print_hashArray
+         @hash_array.each do |element|
+             puts element
+        end
+    end
+
+    def display_hashes(display_fields,group_by,show_total)
+        display_fields=display_fields.split(",")
+        group_by=group_by.upcase
+        total_physics=0
+        total_maths=0
+        total_chemistry=0
+        if display_fields.size==1
+             display_fields[0]=display_fields[0].upcase
+
+
+           puts " |   #{group_by}   |    #{display_fields[0]}      |"
+          if display_fields[0]=="PHYSICS"
+                 @hash_array_groupby.each do |element|
+                      total_physics=total_physics+element[:physics]
+                      puts " |   #{element[element.keys[0]]}   |    #{element[:physics]}      |"
+                    end  # loop end
+                    if show_total.to_s.upcase=="TRUE"
+                    puts " |   Total   |    #{total_physics}      |"
+                  end
+            elsif display_fields[0]=="MATHS"
+                   @hash_array_groupby.each do |element|
+                        total_maths=total_maths+element[:maths]
+                        puts " |   #{element[element.keys[0]]}   |    #{element[:maths]}      |"
+                       end  # loop end
+                        if show_total.to_s.upcase=="TRUE"
+                         puts " |   Total   |    #{total_maths}      |"
+                       end
+            elsif display_fields[0]=="CHEMISTRY"
+                   @hash_array_groupby.each do |element|
+                         total_chemistry=total_chemistry+element[:chemistry]
+                         puts " |   #{element[element.keys[0]]}   |    #{element[:chemistry]}      |"
+                        end  # loop end
+                        if show_total.to_s.upcase=="TRUE"
+                         puts " |   Total   |    #{total_chemistry}      |"
+                       end
+           end  # if-else end
+         elsif display_fields.size==2
+           display_fields[0]=display_fields[0].upcase
```

Comment:
this command not working , shows error
![screenshot from 2016-12-21 17-34-22](https://cloud.githubusercontent.com/assets/24517591/21388782/df7d1148-c7a3-11e6-9cb0-b4b3eb3da77a.png)
r